src/client/ns_turn_msg.c

```diff
@@ -244,23 +244,37 @@ int stun_produce_integrity_key_str(const uint8_t *uname, const uint8_t *realm, c
 		unsigned int keylen = 0;
 		EVP_MD_CTX ctx;
 		EVP_MD_CTX_init(&ctx);
-#if defined EVP_MD_CTX_FLAG_NON_FIPS_ALLOW && !defined(LIBRESSL_VERSION_NUMBER)
+#if defined(OPENSSL_FIPS)
+#if (OPENSSL_VERSION_NUMBER >= 0x30000000L)
+		if (EVP_default_properties_is_fips_enabled(NULL)) {
+			EVP_default_properties_enable_fips(NULL, 1);
+		}
+#else
 		if (FIPS_mode()) {
 			EVP_MD_CTX_set_flags(&ctx,EVP_MD_CTX_FLAG_NON_FIPS_ALLOW);
 		}
-#endif
+#endif //(OPENSSL_VERSION_NUMBER >= 0x30000000L)
+#endif //defined(OPENSSL_FIPS)
+
 		EVP_DigestInit_ex(&ctx,EVP_md5(), NULL);
 		EVP_DigestUpdate(&ctx,str,strl);
 		EVP_DigestFinal(&ctx,key,&keylen);
 		EVP_MD_CTX_cleanup(&ctx);
 #else
 		unsigned int keylen = 0;
 		EVP_MD_CTX *ctx = EVP_MD_CTX_new();
-#if defined EVP_MD_CTX_FLAG_NON_FIPS_ALLOW && ! defined(LIBRESSL_VERSION_NUMBER)
+#if defined(OPENSSL_FIPS)
+#if (OPENSSL_VERSION_NUMBER >= 0x30000000L)
+		if (EVP_default_properties_is_fips_enabled(NULL)) {
```

Comment:
Is this logic correct?    If I understand it correctly based on the openssl doc it says "if (fips enabled) enable fips".   Shouldn't it be the oposite? "if (fips enabled) disable fips"?

Reply:
Yes I see what you mean.
I made a few more mistakes here